libraries/azurerm_resource.rb

```diff
@@ -44,6 +45,34 @@ def assign_fields_with_map(map, struct)
       instance_variable_set("@#{name}", struct.key?(api_name) ? struct[api_name] : nil)
     end
   end
+
+  # Converts the given Azure ID to a hash representation such that:
+  # "/subscriptions/xxx/resourceGroups/my-rg/providers/Microsoft.Compute/disks/my-disk"
+  # becomes {:subscription => 'xxx', :resource_group => 'my-rg', :provider => 'Microsoft.Compute', :disk => 'my-disk'}
+  def id_to_h(azure_id)
+    raise ArgumentError, "`#{azure_id}` is not a valid Azure ID." unless !azure_id.nil? && azure_id.start_with?('/subscriptions')
+
+    begin
+      split = azure_id.split('/').reject!(&:empty?)
+      raise unless !split.empty? && split.length.even?
+      hash = {}
+
+      i = 0
+      counter = 0
+      while counter < split.length/2
+        # Remove plural from API noun.
+        k = split[i].end_with?('s') ? split[i][0..-2] : split[i]
```

Comment:
Are there any cases here where the API resource noun legitimately ends in an 's' e.g. `log_analytics` maybe?  If so I'm wondering if it might be preferable to leave this field unaltered and go with exactly what appears in the fully qualified url.

Reply:
Change added 👍